### PR TITLE
8954: handle potential undefined tab title for sidebar starter brick

### PR DESCRIPTION
**Pull request by fungairino**
In `sidebarStarterBrick.ts`, the code has been refactored to wrap a larger portion of the functionality in a try-catch block to ensure uncaught errors don't result in a stuck sidebar rendering state.

In `sidebarStarterBrickTypes.ts`, the `heading` property in `SidebarConfig` type has been made optional to handle cases where the page editor sets empty string values to undefined. 

Setting an undefined tab title now will be defaulted to an empty string (rather than throwing an error).

### CODE REVIEW

File: src/starterBricks/factory.ts [hunk 1]
```diff
@@ -59,7 +59,5 @@ export function fromJS(definition: StarterBrickDefinitionLike): StarterBrick {
     );
   }
 
-  // TODO: Find a better solution than casting to any
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-argument -- the factory methods perform validation
-  return TYPE_MAP[definition.definition.type](getPlatform(), definition as any);
```
Comment (fungairino) [inline src/starterBricks/factory.ts:64]: An unrelated change - seems like this type is now defined properly.

File: src/starterBricks/sidebar/sidebarStarterBrick.ts [hunk 2]
```diff
@@ -172,37 +172,34 @@ export abstract class SidebarStarterBrickABC extends StarterBrickABC<SidebarConf
   ): Promise<void> {
     // Generate our own run id so that we know it (to pass to upsertPanel)
     const runId = uuidv4();
+    let { heading = "", body } = modComponent.config;
```
Comment (fungairino) [inline src/starterBricks/sidebar/sidebarStarterBrick.ts:175]: this is the quick fix - default heading to empty string if undefined

**Review by grahamlangford — APPROVED** [review]
